### ISSUE TITLE
lib target in Makefile, ubuntu 14.04: curl trouble and --strip-components inconsistency

### ISSUE THREAD

**Issue by scgilardi** [pixie-lang/pixie#306]
When trying to build_with_jit on ubuntu 14.04 I had two problems:
- curl trouble
  - the `curl` available from `apt-get` by default is version `7.35.0` which has trouble downloading the appropriate externals.tar.bz as shown here: https://gist.github.com/scgilardi/c8fa945453c7956d15b6
  - using `curl` version `7.41.0` via MacPorts on Mac OS X has no trouble
  - using `wget` on ubuntu 14.04 has no trouble:
`wget https://github.com/pixie-lang/external-deps/releases/download/1.0/Linux-x86_64.tar.bz2 -O /tmp/externals.tar.bz2`
  - I don't see a clear best solution for this. it might be nice to use wget on ubuntu and curl on the mac
- `--strip-components` inconsistency
  - currently the Mac version of the file that's downloaded as /tmp/externals/tar.bz2 is built with two components before "lib" (and "include")
    - paths begin with "./externals/"
  - in contrast the Linux version (https://github.com/pixie-lang/external-deps/releases/download/1.0/Linux-x86_64.tar.bz2) is built with one component before "lib" (and "include")
    - paths begin with "externals/"
  - when expanding on Linux, there is no lib or include directory produced
  - building and running appears to work, but littering the checkout dir with libraries and include files is clearly not what's intended

**Comment by heyLu** [comment]
The `curl` problem looks like an error on GitHub's side. Can you try again, maybe it went away? (If not, can you post the exact url it's trying to fetch?)

**Comment by scgilardi** [comment]
the download problem (curl 7.35.0 on linux) has persisted for a couple of weeks. I've updated the gist with comments showing "curl -v" output for the unsuccessful (7.35.0/linux) and successful (7.41.0/darwin) cases: https://gist.github.com/scgilardi/c8fa945453c7956d15b6 .

**Comment by heyLu** [comment]
Not sure what is going on. `curl` on Archlinux (7.41.0) works, as does 7.35.0 in a `ubuntu:14.04` docker container.

The `SignatureDoesNotMatch` error from the original gist does not appear in the more detailed log, does that error still occur?

**Comment by scgilardi** [comment]
yes, it ends up as the downloaded contents at /tmp/foo. I added a comment to the gist showing that contents.

I'm running from a vagrant-built vm. I'll see if I can narrow down why your 7.35.0 in an ubuntu:14.04 docker container differs from my 7.35.0 in a vagrant box.

**Comment by scgilardi** [comment]
I built the latest curl on my vm and found that it had the same problem. that led me to look into my dotfiles and it turns out that my ".curlrc" was the culprit it contained:
```
header: "Content-Type:application/json"
header: "Accept:application/json"
```
The curl issue was my fault and the version difference was a red herring.

Thank you very much for your help.

If I can I'll edit the issue down to only the "strip-components" inconsistency.

**Comment by heyLu** [comment]
Ok, then we have a partial happy ending! :) You can leave the description as is, otherwise these comments will be a little confusing. (If you want you can strike out the `curl` part.)

As for the inconsistency in the external-deps tarball, I hope @halgari can help with that. (It should be something in the Makefile task for building it.)

**Comment by halgari** [comment]
I'll take a look.
